.teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt

```diff
@@ -230,6 +231,45 @@ class CIConfigIntegrationTests {
         }
     }
 
+    @Test
+    fun testsAreCorrectlyConfiguredForAllSubProjects() {
+        CIBuildModel().subProjects.filter {
+            !listOf(
+                "soak", // soak test
+                "distributions", // build distributions
+                "docs", // sanity check
+                "architectureTest" // sanity check
+            ).contains(it.name)
+        }.forEach {
+            val dir = getSubProjectFolder(it)
+            assertEquals(it.unitTests, File(dir, "src/test").isDirectory, "${it.name}'s unitTests is wrong!")
+            assertEquals(it.functionalTests, File(dir, "src/integTest").isDirectory, "${it.name}'s functionalTests is wrong!")
+            assertEquals(it.crossVersionTests, crossVersionTestsExistIn(dir), "${it.name}'s crossVersionTests is wrong!")
+        }
+    }
+
+    private fun crossVersionTestsExistIn(dir: File): Boolean {
+        if (File(dir, "src/crossVersionTest").isDirectory) {
+            return true
+        }
+        if (!File(dir, "src/integTest").isDirectory) {
+            return false
+        }
+
+        val keywords = listOf("org.gradle.integtests.fixtures.MultiVersionIntegrationSpec",
+            "org.gradle.integtests.fixtures.MultiVersionSpecRunner")
+        File(dir, "src/integTest").walkBottomUp().forEach { file ->
```

Comment:
@wolfs I'm a little confused here, does "integration tests with MultiVersionRunner` count as "crossVersionTests"?

Reply:
OK the crossVersion means "crossGradleVersion", now I understand.